hazelcast/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java

```diff
@@ -350,7 +355,18 @@ public void run() {
                 if (progressTracker.isMadeProgress()) {
                     idleCount = 0;
                 } else {
-                    idlerLocal.idle(++idleCount);
+                    if (trackers.isEmpty()) {
+                        newTaskletSemaphore.drainPermits();
+                        if (trackers.isEmpty() && !isShutdown) {
+                            try {
+                                newTaskletSemaphore.acquire();
+                            } catch (InterruptedException e) {
+                                logger.fine(e.getMessage());
```

Comment:
I guess the message is `null`.

We should not ignore the interrupt. Jet doesn't use it, so it should never happen - in that case i guess we should log it on the error level and return.
```suggestion
                                logger.severe("Cooperative worker interrupted", e);
                                return;
```